app/services/users/omniauth.rb

```diff
@@ -6,31 +6,37 @@ def initialize(auth)
   end
 
   def get
-    user
+    check_auth
+    @user
   end
 
   private
 
   delegate :provider, :uid, :info, to: :auth
   delegate :email, to: :info
 
-  def user
-    @_user ||= find_user || create_user
+  def find_authorization(user)
+    user.authorizations.where(provider: provider, uid: uid).first
   end
 
-  def find_user
-    User.where(provider: provider, uid: uid, email: email).first
+  def create_authorization(user)
+    user.authorizations.create(provider: provider, uid: uid)
+  end
+
+  def initiate_user
+    @user = User.where(email: email).first || create_user
+  end
+
+  def check_auth
+    initiate_user
+    create_authorization(@user) if find_authorization(@user).blank?
   end
```

Comment:
@sinpai `check_auth` is placed after `create_authorization` and `find_authorization`. Developers are reading code from up to down, so chain of methods should go from up to down:
```ruby
def check_auth
end

def initiate_user
end

def create_authorization
end

def find_authorization
end
```